flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/ExecutionContext.java

```diff
@@ -151,6 +153,9 @@ private ExecutionContext(
 		classLoader = FlinkUserCodeClassLoaders.parentFirst(
 				dependencies.toArray(new URL[dependencies.size()]),
 				this.getClass().getClassLoader());
+		if (!dependencies.isEmpty()) {
+			flinkConfig.set(PipelineOptions.JARS, dependencies.stream().map(URL::toString).collect(Collectors.toList()));
```

Comment:
`ConfigUtils.encodeCollectionToConfig(configuration, PipelineOptions.JARS, jobJars, URL::toString);`

Reply:
https://github.com/apache/flink/pull/10501 aims to fix it.

Reply:
@lirui-apache I think this PR https://github.com/apache/flink/pull/10501 can fix this issue. So suggest reproduce after that PR merged.

Reply:
Sure, I'll wait for #10501 and rebase

Reply:
@lirui-apache #10501 have merged in master and 1.10